### ISSUE TITLE
Sketch plugin error with Sketch 3.5

### ISSUE THREAD

**Issue by DTScorey** [avocode/avocode#609]
IMPORTANT: Please try to describe how did it happen.

-----
Sketch Exporter 2.1.5 running in Sketch 3.5
```
Error: Error: Command failed: 2016-01-26 12:14:18.905 sketchtool[5942:35004339] Error: caught exception whilst suppressing update events: *** -[MSUnarchiver decodeObjectForKey:]: cannot decode object of class (MSImmutableSharedStyleContainer)
2016-01-26 12:14:18.907 sketchtool[5942:35004339] Error: caught exception whilst decoding (version 76):*** -[MSUnarchiver decodeObjectForKey:]: cannot decode object of class (MSImmutableSharedStyleContainer)
Failed to open document file:///Users/corbettmorgan/Downloads/DTS_Login(Final).sketch
((null) MSDocumentDomain:11)


    at /Users/corbettmorgan/…/avocode-exporter/lib/image-exporter.js:1:873
    at ChildProcess.exithandler (child_process.js:651:7)
    at ChildProcess.EventEmitter.emit (events.js:98:17)
    at maybeClose (child_process.js:753:16)
    at Socket.<anonymous> (child_process.js:966:11)
    at Socket.EventEmitter.emit (events.js:95:17)
    at Pipe.close (net.js:465:12)
```

**Comment by rychlis** [comment]
Hi, the Sketch 3.5 update broke our preview image export unfortunately, we're working on a fix that should be ready today.

Until then you can go back to Sketch 3.4.4: http://download.sketchapp.com/sketch-3.4.4.zip

If you already saved sketch files in Sketch 3.5 you will be unable to open it in 3.4.4, for that case you can try to revert to older version using this guide:
0. Make a backup of the latest document in case you made changes since updating Sketch to 3.5
1. Open the document in the new Sketch 3.5
2. In upper menu select File -> Revert To -> Browse All Versions...
3. Select the document from 3.4.x and click on Restore button
4. Close Sketch 3.5 (don't save the document!) and you will be able to open the document in Sketch 3.4.4 now

sorry for the inconvenience!

**Comment by StoneGod** [comment]
I try to use https://github.com/avocode/avocode/blob/master/sketch-extension-beta.md
but it's sync too much time

**Comment by rychlis** [comment]
@StoneGod beta extension is affected as well, we will release a fix right after we finish the update of stable avocode extension

**Comment by rychlis** [comment]
We have just released Sketch Extension 3.0.0-prerelease4 with Sketch 3.5 support. Please download the latest version from https://github.com/avocode/avocode/blob/master/sketch-extension-beta.md#installation--update

We're working on updating the older Sketch Plugin 2 however there has been some further issues with Sketch 3.5 unfortunately. You should be able to use the beta version fine though

**Comment by al1zoin** [comment]
Thanks for that cue. Seems to be an issue with that new Sketch Extension 3.0.0-prerelease4 thought. At least with Sketch Version 3.5.1 (25234). It works up to the point where I am supposed to confirm sync: the confirmation button stays disabled no matter what I do.
![capture d ecran 2016-01-27 a 11 16 56](https://cloud.githubusercontent.com/assets/16922159/12619991/7544a534-c4e8-11e5-8319-3daa38a8a65f.png)

**Comment by rychlis** [comment]
@al1zoin the button will be disabled until you select a project, did you try that?

**Comment by al1zoin** [comment]
Ok, thanks for the answer. I tried Creating a new folder instead since these mock-ups are to be added to a new project, but I couldn't confirm the project's name. Indeed, if I choose an existing project, it enables the sync button. But, as I read above, the synching seems to be quite long.

**Comment by al1zoin** [comment]
Thanks for your time by the way!

**Comment by rychlis** [comment]
@al1zoin good catch, syncing to a new project is impossible now. I've created a separate issue for that: #637 Thanks!

**Comment by al1zoin** [comment]
Ok!

**Comment by designnotdrum** [comment]
@rychlis Are there any detailed instructions on making the new beta sync work? I have tried to install it multiple times with various versions of sketch, back to 3.4.1, and have had zero success in getting it to work.

**Comment by rychlis** [comment]
@designnotdrum could you elaborate a bit? What exactly didn't work? Please report prerelease4 probllem as a new issue or give us an email at team@avocode.com

**Comment by rychlis** [comment]
We've released Avocode Sketch Plugin 3.1.0 which is now the official latest version of the extension and supports both Sketch 3.5.x and 3.4.x. You can get it from here https://manager.avocode.com/downloads/#plugins

**Comment by al1zoin** [comment]
Last release 3.1.0 works well. Thanks! However, if I may say so, it would be nice if the progress bar actually showed the synching progress. Right now, even though I have 3 of my Arboards out of 7 actually synched, the progress bar isn't moving at all until the synch is fully 
![capture d ecran 2016-02-01 a 16 52 39](https://cloud.githubusercontent.com/assets/16922159/12732713/d21cfba6-c904-11e5-95c1-0d372d280e15.png)
completed.
